vllm/model_executor/models/telechat2.py

```diff
@@ -0,0 +1,435 @@
+            config.apply_residual_connection_post_layernorm
+
+    def forward(
+        self,
+        positions: torch.Tensor,
+        hidden_states: torch.Tensor,
+        kv_cache: torch.Tensor,
+        attn_metadata: AttentionMetadata,
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        residual = hidden_states
+        layernorm_output = self.input_layernorm(hidden_states)
+        attn_outputs = self.self_attention(
+            positions=positions,
+            hidden_states=layernorm_output,
+            kv_cache=kv_cache,
+            attn_metadata=attn_metadata,
+        )
+        attn_outputs = residual + attn_outputs
+        residual = attn_outputs
+        layernorm_output = self.post_attention_layernorm(attn_outputs)
+        output = residual + self.mlp(layernorm_output)
+        return output
+
+
+class TeleChat2Model(nn.Module):
+    
+    def __init__(
+        self,
+        config: PretrainedConfig,
+        cache_config: Optional[CacheConfig] = None,
+        quant_config: Optional[QuantizationConfig] = None,
+        lora_config: Optional[LoRAConfig] = None,
+    ) -> None:
+        super().__init__()
+        self.config = config
+        self.padding_idx = config.pad_token_id
+        self.vocab_size = config.vocab_size
+        self.word_embeddings = VocabParallelEmbedding(
+            self.vocab_size,
+            config.hidden_size,
+            org_num_embeddings=config.vocab_size,
+        )
+        self.h = nn.ModuleList(
+           [
+                TeleChat2DecoderLayer(config=config,
+                                  cache_config=cache_config,
+                                  quant_config=quant_config)
+                for _ in range(self.config.num_hidden_layers)
+            ] 
+        )
+        
+        self.ln_f = RMSNorm(config.hidden_size, eps=config.rms_norm_eps)
+
+    def get_input_embeddings(self, input_ids: torch.Tensor) -> torch.Tensor:
+        return self.word_embeddings(input_ids)
+
+    def forward(
+        self,
+        input_ids: Optional[torch.Tensor],
+        positions: torch.Tensor,
+        kv_caches: List[torch.Tensor],
+        attn_metadata: AttentionMetadata,
+    ) -> Union[torch.Tensor, IntermediateTensors]:
+        
+        hidden_states = self.get_input_embeddings(input_ids)
+       
+        for i in range(self.config.num_hidden_layers):
+            layer = self.h[i]
+            hidden_states = layer(
+                positions,
+                hidden_states,
+                kv_caches[i],
+                attn_metadata,
+            )
+
+        hidden_states = self.ln_f(hidden_states)
+        return hidden_states
+
+
+class TeleChat2ForCausalLM(nn.Module, SupportsLoRA):
+    
+    def __init__(
+        self,
+        config: PretrainedConfig,
+        cache_config: Optional[CacheConfig] = None,
+        quant_config: Optional[QuantizationConfig] = None,
+        lora_config: Optional[LoRAConfig] = None,
+    ) -> None:
```

Comment:
Please replace these arguments with `vllm_config`,The same applies to `TeleChat2Model`. You can refer to the `Llama` implementation